opentelemetry/proto/profiles/v1/profiles.proto

```diff
@@ -244,6 +247,10 @@ message Mapping {
   // with high probability. E.g., for binaries generated by GNU tools,
   // it could be the contents of the .note.gnu.build-id field. Index into string table
   uint32 build_id_index = 5;
+  // A string that uniquely identifies a particular binary with high probability.
+  // It is meant to work around deficiencies in the GNU Build ID that may compromise
+  // uniqueness / correlatability. Index into string table.
+  uint32 file_id_index = 6;
```

Comment:
The GNU Build ID may be missing (Alpine) or not be usable for uniquely identifying binaries. It may also be set to random bytes which makes correlation across builds impossible as its value is not derived from the file. 

An alternative is to fully specify a hashing scheme for a new `file_id` that doesn't suffer from these issues. Thomas Dullien has [suggested](https://github.com/open-telemetry/oteps/pull/237#issuecomment-1717541362) hashing the ELF header and a few deterministically selected executable pages. We could then either replace `build_id_index` with `file_id_index`, or if we still want to keep `build_id_index` around with its original semantics (e.g. containing GNU Build ID), introduce a separate field as shown here.